pulsar/producer_partition.go

```diff
@@ -373,10 +390,14 @@ func (p *partitionProducer) GetBuffer() internal.Buffer {
 	return b
 }
 
-func (p *partitionProducer) ConnectionClosed() {
+func (p *partitionProducer) ConnectionClosed(closeProducer *pb.CommandCloseProducer) {
```

Comment:
I see `ConnectionClosed(nil)` could be called in `connection.go`, should you perform the null check here?

Reply:
Good catch, fixed!